bst-lowest-common-ancestor/solution.cc

```diff
@@ -0,0 +1,43 @@
+/*https://www.hackerrank.com/challenges/binary-search-tree-lowest-common-ancestor/copy-from/17942742*/
+
+/*
+Node is defined as 
+
+typedef struct node
+{
+   int data;
+   node * left;
+   node * right;
+}node;
+*/
+
+node * lca(node * root, int v1, int v2)
+{   
+    if(root == NULL)
+    {
+        return NULL;
+    }
+    
+    int smallV = min(v1, v2);
+    int largeV = max(v1, v2);
+    int rootV = root->data;
+    
+    if(rootV >= smallV && rootV <= largeV)
+    {
+        return root;
+    }
+    else if(rootV > largeV)
+    {
+        if(root->left == NULL){
+            return root;
+        }
+        return lca(root->left, smallV, largeV);
+    }else
+    {
+        if(root->right == NULL)
+        {
+            return root;
+        }
+        return lca(root->right, smallV, largeV);
+    }
+}
```

Comment:
You should change your editor to automatically fix the 'no newline at EOF' problem: https://forum.sublimetext.com/t/make-saving-newline-at-eof-the-installation-default/9842

Otherwise AMAZINGGGGg